vignettes/estimate_infections_options.Rmd

```diff
@@ -0,0 +1,667 @@
+#>         warmup sample
+#> chain:1 36.415 38.937
+#> chain:2 29.110 34.879
+#> chain:3 34.784 35.283
+#> chain:4 33.756 50.211
+# summary plot
+plot(no_delay)
+```
+
+![plot of chunk no_delays](figure/no_delays-1.png)
+
+## Non-parametric infection model
+
+The package also includes a non-parametric infection model.
+This runs much faster but does not use the renewal equation to generate infections.
+Because of this none of the options defining the behaviour of the reproduction number are available in this case, limiting user choice and model generality.
+It also means that the model is questionable for forecasting, which is why were here set the predictive horizon to 0.
+
+
+```r
+non_parametric <- estimate_infections(reported_cases,
+  generation_time = generation_time_opts(generation_time),
+  delays = delay_opts(delay),
+  rt = NULL,
+  backcalc = backcalc_opts(),
+  horizon = 0
+)
+#> DEBUG [2023-09-27 14:53:18] estimate_infections: Running in exact mode for 2000 samples (across 4 chains each with a warm up of 250 iterations each) and 73 time steps of which 0 are a forecast
+#> Warning in validityMethod(object): The following variables have undefined
+#> values: gt_rev_pmf[1],The following variables have undefined values:
+#> gt_rev_pmf[2],The following variables have undefined values: gt_rev_pmf[3],The
+#> following variables have undefined values: gt_rev_pmf[4],The following
+#> variables have undefined values: gt_rev_pmf[5],The following variables have
+#> undefined values: gt_rev_pmf[6],The following variables have undefined values:
+#> gt_rev_pmf[7],The following variables have undefined values: gt_rev_pmf[8],The
+#> following variables have undefined values: gt_rev_pmf[9],The following
+#> variables have undefined values: gt_rev_pmf[10],The following variables have
+#> undefined values: gt_rev_pmf[11],The following variables have undefined values:
+#> gt_rev_pmf[12],The following variables have undefined values:
+#> gt_rev_pmf[13],The following variables have undefined values:
+#> gt_rev_pmf[14],The following variables have undefined values: gt_rev_pmf[15].
+#> Many subsequent functions will not work correctly.
+# summarise results
+summary(non_parametric)
+#>                                  measure                  estimate
+#> 1: New confirmed cases by infection date       2333 (1759 -- 3070)
+#> 2:        Expected change in daily cases                Decreasing
+#> 3:            Effective reproduction no.        0.9 (0.78 -- 0.99)
+#> 4:                        Rate of growth -0.023 (-0.049 -- -0.002)
+#> 5:          Doubling/halving time (days)         -30 (-340 -- -14)
+# elapsed time (in seconds)
+get_elapsed_time(non_parametric$fit)
+#>         warmup sample
+#> chain:1  2.953  0.957
+#> chain:2  3.467  0.933
+#> chain:3  3.384  0.857
+#> chain:4  3.659  0.963
+# summary plot
+plot(non_parametric)
+```
+
+![plot of chunk nonparametric](figure/nonparametric-1.png)
+
+# Running the model in production mode
```

Comment:
This feels a little like a different vignette

Reply:
good idea - will pull it out